### PR TITLE
chore(go.mod): update juju dependency to 3.6.0

### PR DESCRIPTION
**Pull request by alesstimec**
## Description

Updates juju dependency to 3.6.0.

## Type of change

- Other (please describe)

Updates juju dependency to 3.6.0.

### CODE REVIEW

File: internal/provider/main_test.go [hunk 1]
```diff
@@ -90,6 +90,6 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		panic(err)
 	} else {
-		m.Run()
+		os.Exit(m.Run())
```
Comment (ale8k) [inline internal/provider/main_test.go:93]: Why is this necessary now? We're exiting anyway?

**Review by ale8k — APPROVED** [review]
lgtm

File: internal/provider/main_test.go (hunk 1 above)
Comment (alesstimec) [reply internal/provider/main_test.go:93]: linter was complaining

**Review by anvial — APPROVED** [review]
I'm OK with update.

I've checked `make build` and deployed some stuff with Juju controller.
